src/ddb/ddb_vos.c

```diff
@@ -1636,3 +1636,59 @@ dv_sync_smd(dv_smd_sync_complete complete_cb, void *cb_args)
 
 	return rc;
 }
+
+struct vea_cb_args {
+	dv_vea_extent_handler	 vca_cb;
+	void			*vca_cb_args;
+};
+
+static int
+vea_free_extent_cb(daos_handle_t ih, d_iov_t *key, d_iov_t *val, void *cb_arg)
+{
+	struct vea_cb_args	*args = cb_arg;
+	struct vea_free_extent	*vfe;
+
+	vfe = (struct vea_free_extent *)val->iov_buf;
+
+	if (args->vca_cb)
+		return args->vca_cb(args->vca_cb_args, vfe);
+
+	return 0;
+}
+
+int
+dv_enumerate_vea(daos_handle_t poh, dv_vea_extent_handler cb, void *cb_arg)
+{
+	struct vea_cb_args	 args = {.vca_cb = cb, .vca_cb_args = cb_arg};
+	struct vos_pool		*pool;
+	struct vea_space_info	*vsi;
+	int			 rc;
+
+	pool = vos_hdl2pool(poh);
+	vsi = pool->vp_vea_info;
+	if (vsi == NULL)
+		return -DER_NONEXIST;
+
+	rc = dbtree_iterate(vsi->vsi_md_free_btr, 0, false, vea_free_extent_cb, &args);
```

Comment:
Hmm, I was thinking that DDB is just like other server components. Is it supposed to be able to access all these internal data structures defined in vos_internal.h and vea_internal.h?

Should we try to introduce minimum exposing to satisfy DDB and other potential common use cases, or should we expose everything to DDB? I think the former way is easier for code maintenance.

Reply:
I updated it to remove the dependency on vea_internal.h by adding a free enumeration api to vea. @NiuYawei let me know what you think.